pal/screen-adapter/web/screen-adapter.ts

```diff
@@ -71,36 +107,7 @@ class ScreenAdapter extends EventTarget {
             return;
         }
         this._orientation = value;
-        if (!this._gameFrame) {
-            return;
-        }
-        const width = window.innerWidth;
-        const height = window.innerHeight;
-        const isBrowserLandscape = width > height;
-        const needToRotateDocument = systemInfo.isMobile && systemInfo.os === OS.ANDROID
-            && ((isBrowserLandscape && value & Orientation.PORTRAIT) || (!isBrowserLandscape && value & Orientation.LANDSCAPE));
-        if (needToRotateDocument) {
-            this.isFrameRotated = true;
-            this._gameFrame.style['-webkit-transform'] = 'rotate(90deg)';
-            this._gameFrame.style.transform = 'rotate(90deg)';
-            this._gameFrame.style['-webkit-transform-origin'] = '0px 0px 0px';
-            this._gameFrame.style.transformOrigin = '0px 0px 0px';
-            this._gameFrame.style.margin = `0 0 0 ${width}px`;
-            this._gameFrame.style.width = `${height}px`;
-            this._gameFrame.style.height = `${width}px`;
-            this._resizeFrame(new Size(height, width));
-        } else {
-            this.isFrameRotated = false;
-            this._gameFrame.style['-webkit-transform'] = 'rotate(0deg)';
-            this._gameFrame.style.transform = 'rotate(0deg)';
-            // TODO
-            // this._gameFrame.style['-webkit-transform-origin'] = '0px 0px 0px';
-            // this._gameFrame.style.transformOrigin = '0px 0px 0px';
-            this._gameFrame.style.margin = '0px';
-            this._gameFrame.style.width = `${width}px`;
-            this._gameFrame.style.height = `${height}px`;
-            this._resizeFrame(new Size(width, height));
-        }
+        this._updateFrameState();
```

Comment:
looks like immediate frame setting is removed, how to ensure a dynamic set to the orientation works well ?

Reply:
**Immediate Frame Setting** is seperate into two part: **Update Frame State** and **Reszie Frame**
- We need to **update frame state** only when setting orientation or system orientation changed 
- We need to **resize frame** when init screen, setting window size, system window resize or system orientation changed